framework/src/endpoint.ts

```diff
@@ -21,7 +21,18 @@ export const isReservedEndpointFunction = (key: string): boolean =>
 
 export const getEndpointHandlers = (endpoint: Record<string, unknown>): EndpointHandlers => {
 	const endpointHandlers: EndpointHandlers = {};
-	for (const key of Object.getOwnPropertyNames(Object.getPrototypeOf(endpoint))) {
+	let isBaseEndpoint;
+
+	// For endpoints which extend other endpoints, go deeper in the prototype chain
+	do {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, no-param-reassign
+		endpoint = Object.getPrototypeOf(endpoint);
+		isBaseEndpoint =
+			(Object.getPrototypeOf(Object.getPrototypeOf(endpoint)) as Record<string, unknown>)
+				.constructor.name === Object.name;
+	} while (!isBaseEndpoint);
+
+	for (const key of Object.getOwnPropertyNames(endpoint)) {
```

Comment:
i think in this way, it's not possible to get both base endpoint and some endpoint on the extended class?